backend/migrations/dataall_migrations/migrationmanager.py

```diff
@@ -0,0 +1,67 @@
+import os
+from collections import deque
+from typing import Deque
+from migrations.dataall_migrations.versions.initial import InitMigration
+from migrations.dataall_migrations.base_migration import BaseDataAllMigration
+
+import logging
+
+logger = logging.getLogger()
+logger.setLevel(os.environ.get('LOG_LEVEL', 'INFO'))
+
+
+class MigrationManager:
+    def __init__(self, key='0'):
+        self.current_migration = InitMigration
+        self.previous_migrations: Deque[BaseDataAllMigration] = deque()
+        while self.current_migration.key() != key:
+            self.previous_migrations.append(self.current_migration)
+            self.current_migration = self.current_migration.next_migration()
+        self.executed_upgrades: Deque[BaseDataAllMigration] = deque()
+        self.executed_downgrades: Deque[BaseDataAllMigration] = deque()
+
+    def upgrade(self, target_key=None):
+        if self.current_migration.key() == target_key or self.current_migration.next_migration() is None:
+            logger.info('Data-all version is up to date')
+            return self.current_migration.key()
+
+        logger.info(
+            f"Upgrade from {self.current_migration.key()} to {target_key if target_key is not None else 'latest'}"
+        )
+        while self.current_migration is not None and self.current_migration.key() != target_key:
+            try:
+                self.executed_upgrades.append(self.current_migration)
+                logger.info(f'Applying migration {self.current_migration.name()}')
+                self.current_migration.up()
+                logger.info(f'Migration {self.current_migration.name()} completed')
+                self.current_migration = self.current_migration.next_migration()
+            except Exception as e:
+                # Something went wrong revert
+                logger.info(f'An error occurred while applying the migration.{e}.')
+                while migration := self.executed_upgrades.pop():
+                    migration.down()
+                return False
+        logger.info('Upgrade completed')
+        return self.executed_upgrades.pop().key()
+
+    def downgrade(self, target_key='0'):
```

Comment:
this is unused, thus I don't see why having it in the codebase

Reply:
I would save it for local testing purposes.